File: distributed-exceptions/src/main/java/io/axoniq/distributedexceptions/DistributedExceptionsApplication.java
package io.axoniq.distributedexceptions;

import io.axoniq.distributedexceptions.command.ExceptionWrappingHandlerInterceptor;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Profile;

@SpringBootApplication
public class DistributedExceptionsApplication {

    public static void main(String[] args) {
        SpringApplication.run(DistributedExceptionsApplication.class, args);
    }

    @Bean
    @Profile("command")
    public ExceptionWrappingHandlerInterceptor exceptionWrappingHandlerInterceptor() {
        return new ExceptionWrappingHandlerInterceptor();
    }
}

File: distributed-exceptions/src/main/kotlin/io/axoniq/distributedexceptions/DistributedExceptionsApplication.kt
package io.axoniq.distributedexceptions

import io.axoniq.distributedexceptions.command.ExceptionWrappingHandlerInterceptor
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Profile

/**
 * @author Stefan Andjelkovic
 */
@SpringBootApplication
class DistributedExceptionsApplication {
    @Bean
    @Profile("command")
    fun exceptionWrappingHandlerInterceptor() = ExceptionWrappingHandlerInterceptor()
}

fun main(args: Array<String>) {
    runApplication<DistributedExceptionsApplication>(*args)
}
